src/Middleware/SpaServices.Extensions/src/AngularCli/AngularCliMiddleware.cs

```diff
@@ -64,7 +67,7 @@ internal static class AngularCliMiddleware
         }
 
         private static async Task<AngularCliServerInfo> StartAngularCliServerAsync(
-            string sourcePath, string scriptName, string pkgManagerCommand, int portNumber, ILogger logger)
+            string sourcePath, string scriptName, string pkgManagerCommand, int portNumber, ILogger logger, CancellationToken applicationStoppingToken)
```

Comment:
This design seems reasonable and good to me, but I don't have a background with `IHostApplicationLifetime`, @SteveSandersonMS would you know if there are any pitfalls here, or know who might know?

Reply:
`IHostApplicationLifetime` sounds like a totally reasonable thing to use for this. AFAIK it's the primary thing we have that signals when an application is shutting down.

It's not absolutely bulletproof (if the application dies due to an unhandled exception or being forcibly killed then I don't think it's guaranteed to notify) but this should still be an improvement.